scripts/abi_to_rc/abi_to_rc.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+# By: Jon-Eric Cook
+# Github: @joneric
+
+import sys
+import os
+import json
+
+# global variables
+actions = []
+inputs = {}
+types = {}
+
+# validates passed in arguments
+def validate_input_arguments():
+    # exit if less than 4 arguments are given
+    if len(sys.argv) < 4:
+        print("ERROR: missing arguments")
+        print("-- Expected format --")
+        print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+        sys.exit(0)
+
+    # exit if arguments are ordered incorrectly
+    # exit if passed in files can't be found
+    elif len(sys.argv) == 4:
+        if ".abi" not in sys.argv[1]:
+            print("ERROR: first argument should be a .abi file")
+            print("-- Expected format --")
+            print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+            sys.exit(0)
+        elif not os.path.isfile(sys.argv[1]):
+            print("ERROR: %s could not be found" % sys.argv[1])
+            sys.exit(0)
+        elif "rc-overview-template.md" != sys.argv[2]:
+            print("ERROR: second argument should be the 'rc-overview-template.md' file")
+            print("-- Expected format --")
+            print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+            sys.exit(0)
+        elif not os.path.isfile(sys.argv[2]):
+            print("ERROR: %s could not be found" % sys.argv[2])
+            sys.exit(0)
+        elif "rc-action-template.md" != sys.argv[3]:
+            print("ERROR: third argument should be the 'rc-action-template.md' file")
+            print("-- Expected format --")
+            print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+            sys.exit(0)
+        elif not os.path.isfile(sys.argv[3]):
+            print("ERROR: %s could not be found" % sys.argv[3])
+            sys.exit(0)
+    else:
+        print("ERROR: too many arguments")
+        print("-- Expected format --")
+        print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+        sys.exit(0)
```

Comment:
Error exit values should always be non-zero.  That way different non-zero exit values can mean different, specific error conditions, which makes automated scripting easier.  All Unix command line tools follow this convention, including all of the test code in the repository.  CTest looks for non-zero results and treats them as errors, while passing all zero return codes.